cdap-api/src/main/java/co/cask/cdap/api/flow/AbstractFlow.java

```diff
@@ -23,27 +23,19 @@
 /**
  * This abstract class provides a default implementation of {@link Flow} methods for easy extension.
  */
-public class AbstractFlow implements Flow {
+public abstract class AbstractFlow implements Flow {
 
   private FlowConfigurer configurer;
 
-  @Deprecated
-  @Override
-  public final FlowSpecification configure() {
-    return null;
-  }
-
   public final void configure(FlowConfigurer configurer) {
     this.configurer = configurer;
     configureFlow();
   }
 
   /**
-   * Configures the flow.
+   * Configure the flow.
    */
-  protected void configureFlow() {
-
-  }
+  protected abstract void configureFlow();
```

Comment:
We had a configure method in the previous API that used to return FlowSpecification. So we had to introduce configureFlow method. We can introduce configure method in (3.2) and deprecate this method.